### PR TITLE
Linking libcurl

### PR DESCRIPTION
**Pull request by taketin**
I'll use `curl` to upload cartridge to our server.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,6 +21,7 @@ INCLUDES= \
 	-I$(3RD_PARTY)/wren-0.1.0/src/include \
 	-I$(3RD_PARTY)/moonscript \
 	-I$(3RD_PARTY)/fennel \
+	-I$(3RD_PARTY)/curl \
```
Comment (ericlai616) [inline Makefile:24]: The 3rd-party submodule doesn't contain `curl`, so this might not work... 🤔

Comment (taketin) [reply Makefile:24]: You should do `submodule update` I think.

Comment (ericlai616) [reply Makefile:24]: Memo:
I got this error for `submodule update`
```
error: Server does not allow request for unadvertised object 82bacbf88361c77455cb8b3052d3c07372be13b1
Fetched in submodule path '3rd-party', but it did not contain 82bacbf88361c77455cb8b3052d3c07372be13b1. Direct fetching of that commit failed.
```

It seems that the current `3rd-party` submodule is using the original one.
Perhaps we need to:
- fork the `3rd-party` submodule in `tic80hackathon`
- set the forked `3rd-party` to be this `TIS-80`'s submodule
- add curl dependency in the forked one

**Comment by taketin** [comment]
We should execute the following commands.

```
$ git submodule sync
$ git submodule update --init --recursive
```

Everything all right 🍻

**Comment by taketin** [comment]
FYI: We should use `5b73d39` commit right now because if we use `SDL2-2.0.8` then cause build error (`SDL2-2.0.7` is stable)

**Comment by ericlai616** [comment]
>git update --init --recursive

`git submodule update --init --recursive`?

**Review by ericlai616 — APPROVED** [review]
Finally it can be pulled and built in one shot! LGTM 👍

**Pull request merged**